File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第7章装箱问题.py
#装箱问题  lanqiaoOJ 763

dp = [0]*20010
w = [0]*40
V = int(input()) 
n = int(input())
for i in range(1, n+1):   w[i] = int(input())
for i in range(1,n+1):
    for j in range (V,w[i]-1,-1):
        dp[j] =max(dp[j], dp[j-w[i]]+w[i])
print(V-dp[V]) 

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第7章装箱问题.cpp
//装箱问题  lanqiaoOJ 763

#include<bits/stdc++.h>
using namespace std;
int dp[20010];
int w[40];
int main(){
    int V,n;   scanf("%d%d",&V,&n);
    for(int i=1;i<=n;i++) scanf("%d",&w[i]);
    for(int i=1; i<=n; i++)
        for(int j=V; j>=w[i]; j--)
              dp[j] =max(dp[j], dp[j-w[i]]+w[i]);
    printf("%d\n",V-dp[V]);
}
